### PR TITLE
tilt: ^T toggles log timestamps

### PR DESCRIPTION
**Pull request by landism**
Behavior:
^T toggles timestampped logging
While it's on, all lines are preceded by timestamps.
There is no retroactive adding of timestamps to old lines, and if you turn timestamps back off, the timestamps remain on the old lines.
It's possible at some point we'll save logs with more structure and are able to apply this retroactively, but that didn't seem worth tackling right now.

Leaving this as kind of a hidden feature for now. Filed https://app.clubhouse.io/windmill/story/1798/add-documentation-for-rarer-shortcut-keys to make it discoverable.

It's pretty ugly at default terminal width:
image

1. The timestamps themselves take up a lot of horizontal space, inducing wrapping (ideally, we could make those lines after, e.g., "Building spoonerisms" dynamically grow depending on timestamps, but whatever).
2. Since vigoda logs contain timestamps already, those lines now have two timestamps, which is just a bunch of wasted space. Ditto for Tiltfile logs.

### CODE REVIEW

File: internal/engine/buildcontroller.go [hunk 1]
```diff
@@ -220,9 +220,9 @@ func (c *BuildController) logBuildEntry(ctx context.Context, entry buildEntry, c
 
 	l := logger.Get(ctx)
 	if firstBuild {
-		p := logger.Blue(l).Sprintf("\n──┤ Building: ")
+		p := logger.Blue(l).Sprintf("──┤ Building: ")
```
Comment (landism) [inline internal/engine/buildcontroller.go]: There's a bug in this PR (and also the existing PrefixedWriter) - if something logs a colored newline, that color will also affect that next line's timestamp/prefix.

This is annoying enough to fix and low enough impact (colored prefixes!) that I figured band-aiding it for now is fine.

**Comment by maiamcc** [comment]
Hmm so this won't work with `--hud=false`, correct? how much do we care?

File: internal/model/log.go [hunk 2]
```diff
@@ -27,19 +32,63 @@ func (l Log) Empty() bool {
...
+	}
+
+	content = append(content, b...)
+
+	content = ensureMaxLength(content)
+
+	return Log{content}
+}
+
+func addTimestamps(bs []byte) []byte {
```
[the remark this answers is not in the document]
Comment (landism) [reply internal/model/log.go]: Yeah, I was feeling that here, but didn't want to take on the potential perf hit for this potentially niche feature.

**Comment by landism** [comment]
> Hmm so this won't work with `--hud=false`, correct? how much do we care?

Correct. Do you think we should care? Do we know of users using `--hud=false`, or is it mostly just us during development?

**Comment by maiamcc** [comment]
> Correct. Do you think we should care? Do we know of users using --hud=false, or is it mostly just us during development?

afaik, just us, so go ahead and land and we can figure out a way to add timestamps for `--hud=false` if/when we need it.

**Pull request merged**